libraries/commerce/product/selectors/product.js

```diff
@@ -453,25 +465,64 @@ export const getProductMetadata = createSelector(
 );
 
 /**
- * Indicates whether a product is a base product or not
  * @param {Object} state The current application state.
- * @param {string} productId A product id.
+ * @param {object} props The component props.
  * @return {boolean|null}
  */
 export const isBaseProduct = createSelector(
-  getProductById,
-  (product) => {
-    if (!product.productData || product.isFetching) {
+  getProducts,
+  (state, props) => props.id,
+  (products, productId) => {
+    if (!productId) {
+      return null;
+    }
+
+    if (!products[productId] || !products[productId].productData) {
+      return null;
+    }
+
+    return products[productId].productData.baseProductId === null;
+  }
+);
+
+/**
+ * @param {Object} state The current application state.
+ * @param {object} props The component props.
+ * @return {boolean|null}
+ */
+export const getBaseProductId = createSelector(
+  getProducts,
+  (state, props) => props.id,
+  (products, productId) => {
+    if (!productId) {
+      return null;
+    }
+
+    if (!products[productId] || !products[productId].productData) {
+      return null;
+    }
+
+    if (products[productId].productData.baseProductId) {
+      return products[productId].productData.baseProductId;
+    }
+
+    return productId;
+  }
+);
+
+/**
+ * @param {Object} state The current application state.
+ * @param {object} props The component props.
+ * @return {boolean|null}
```

Comment:
It is not returning a boolean. You return the `productId` or null. So it is a `string`.

Reply:
Fixed